app/views/users/edit.html.erb

```diff
@@ -59,7 +60,11 @@
 
     <div>
       <input type="submit" class="btn btn-primary" value="Save" />
-      <%= link_to "Cancel", cancel_edit_user_path, type: "button", class: "btn btn-default" %>
+      <%= link_to "Cancel", cancel_edit_user_path, class: "btn btn-default" %>
+
+      <% if !@user.deleted? && current_user.can_delete_user? %>
+        <%= link_to "Delete", user_path(@user), method: :delete, class: "btn btn-danger pull-right", data: confirm(title: "Deleting User: #{@user.name}") %>
```

Comment:
Undelete would just be inviting to another organization, so I don't think it is really applicable.